### PR TITLE
Enable Strict Concurrency mode

### PR DESCRIPTION
**Pull request by emildzwonek**
- Enable strict concurrency mode
- Bump minimal OS versions (required by swift-tools-version:5.10)
- Tweak codebase to build with strict concurrency mode without warnings

### CODE REVIEW

File: Sources/URLSessionDecodable/URLSessionDecodable.swift [hunk 1]
```diff
@@ -60,23 +60,23 @@ extension URLSession {
         method: HTTPMethod,
         parameters: ParametersEncoding?,
         headers: HTTPHeaders?,
-        decoder: DataDecoder
+        decoder: @autoclosure @escaping @Sendable () -> DataDecoder
```
Comment (pwszebor-vimn) [inline Sources/URLSessionDecodable/URLSessionDecodable.swift:63]: does it have to be escaping here?

File: Sources/URLSessionDecodable/URLSessionDecodable.swift [hunk 2]
```diff
@@ -60,23 +60,23 @@ extension URLSession {
...
+        return try Self.handle(response: httpResponse, data: data, decoder: decoder(), url: url).get()
     }
 
     // MARK: - Private
 
     private static func handle<T: Decodable>(
         response: HTTPURLResponse,
         data: Data,
-        decoder: DataDecoder,
+        decoder: @autoclosure @escaping @Sendable () -> DataDecoder,
```
Comment (pwszebor-vimn) [inline Sources/URLSessionDecodable/URLSessionDecodable.swift:79]: and here

File: Sources/URLSessionDecodable/URLSessionDecodable.swift (hunk 1 above)
Comment (emildzwonek) [reply Sources/URLSessionDecodable/URLSessionDecodable.swift]: Good catch, removed

File: Sources/URLSessionDecodable/URLSessionDecodable.swift (hunk 2 above)
Comment (emildzwonek) [reply Sources/URLSessionDecodable/URLSessionDecodable.swift]: Good catch, removed

**Review by pwszebor-vimn — APPROVED** [review]

File: Package.swift [hunk 3]
```diff
@@ -1,30 +1,35 @@
...
 
 let package = Package(
     name: "URLSessionDecodable",
     platforms: [
-        .macOS(.v10_13),
-        .iOS(.v11),
-        .tvOS(.v11),
+        .macOS(.v13),
+        .iOS(.v12),
+        .tvOS(.v12),
```
Comment (laurentiuionescu-viacom) [inline Package.swift:10]: why not V15?

**Review by laurentiuionescu-viacom — APPROVED** [review]

**Review by laurentiuionescu-viacom — APPROVED** [review]

**Review by vladkleshchanka — APPROVED** [review]

File: Package.swift (hunk 3 above)
Comment (emildzwonek) [reply Package.swift:10]: Just bumped it to the oldest recommended version. We won't get any benefit from v15 here, v16 would make a difference

**Pull request merged**